### PR TITLE
Make online tests default skip and opt-in

### PR DESCRIPTION
**Pull request by mtreinish**
### Summary

Right now there are a few tests in the terra's unittests that require
being able to talk to ibmq services for verifying functionality that
uses ibmq. However, to run these tests additional manual setup is
required. First the qiskit-ibmq-provider package must be installed, then
ibmq testing credentials must be configured. Prior to this patch it was
assumed both of these manually setup conditions were always true and the
online tests are run and will fail because the manual setup wasn't done.
This commit changes the behavior to make online tests opt-in and skip if
the setup isn't done. This is a more graceful method to handle optional
steps that aren't done, especially to ensure that unittests always pass
from a fresh clone.

### Details and comments

### CODE REVIEW

**Comment by mtreinish** [comment]
The other thing I was debating (because of #2559 ) is changing the test credentials used to only use env variable set credentials instead of defaulting to qiskitrc credentials. So we only run with explicitly set credentials instead of using the normal credentials.

File: .github/CONTRIBUTING.rst [hunk 1]
```diff
@@ -195,33 +195,23 @@ account. To set this up please go to this
...
+| ``skip_online`` | Skips tests that require remote requests (also, no mocked information is used). Does not require user credentials. | ``False`` |
++-----------------+--------------------------------------------------------------------------------------------------------------------+-----------+
+| ``run_slow``    | It runs tests tagged as *slow*.                                                                                    | ``False`` |
++-----------------+--------------------------------------------------------------------------------------------------------------------+-----------+
 
 It is possible to provide more than one option separated with commas.
-The order of precedence in the options is right to left. For example,
-``QISKIT_TESTS=skip_online,rec`` will set the options as
-``skip_online == False`` and ``rec == True``.
+The order of precedence in the options is right to left.
```
Comment (kdk) [inline .github/CONTRIBUTING.rst]: Is this still necessary now that we only have `skip_online` and `run_slow`?

File: .github/CONTRIBUTING.rst [hunk 2]
```diff
@@ -195,33 +195,23 @@ account. To set this up please go to this
 `page <https://quantumexperience.ng.bluemix.net/qx/account/advanced>`_  and
 register an account.
 
-By default, and if there is no user credentials available, the tests that
-require online access are run with recorded (mocked) information. This is, the
-remote requests are replayed from a ``test/cassettes`` and not real HTTP
-requests is generated. If user credentials are found, in that cases it use them
-to make the network requests.
+By default, and if there is no user credentials available.  If user credentials
```
Comment (kdk) [inline .github/CONTRIBUTING.rst]: Partial sentence.

File: .github/CONTRIBUTING.rst [hunk 3]
```diff
@@ -195,33 +195,23 @@ account. To set this up please go to this
...
-+-------------------+--------------------------------------------------------------------------------------------------------------------+-----------------------+--------------------------------------------------+
-| ``rec``           | It records the remote requests. It requires user credentials.                                                      | ``False``             | ``skip_online = False``                          |
-|                   |                                                                                                                    |                       | ``run_slow = False``                             |
-+-------------------+--------------------------------------------------------------------------------------------------------------------+-----------------------+--------------------------------------------------+
++-----------------+--------------------------------------------------------------------------------------------------------------------+-----------+
```
Comment (kdk) [inline .github/CONTRIBUTING.rst]: Can the `uc_available` comment above be removed?

File: .github/CONTRIBUTING.rst (hunk 1 above)
Comment (mtreinish) [reply .github/CONTRIBUTING.rst]: Not really, I'll just remove it.

File: .github/CONTRIBUTING.rst (hunk 3 above)
Comment (mtreinish) [reply .github/CONTRIBUTING.rst]: Yep

File: .github/CONTRIBUTING.rst (hunk 2 above)
Comment (mtreinish) [reply .github/CONTRIBUTING.rst]: Oops, I'll update it. I got a bit too greedy with the removals

**Pull request merged**